src/main/java/mate/academy/rickandmorty/service/CharacterServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package mate.academy.rickandmorty.service;
+
+import jakarta.annotation.PostConstruct;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+import mate.academy.rickandmorty.model.external.CharacterResultDto;
+import mate.academy.rickandmorty.model.internal.Character;
+import mate.academy.rickandmorty.repo.CharacterRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CharacterServiceImpl implements CharacterService {
+    private final CharacterRepository characterRepository;
+    private final RickAndMortyClient dataLoadOnStartup;
+
+    public CharacterServiceImpl(CharacterRepository characterRepository,
+                                RickAndMortyClient dataLoadOnStartup) {
+        this.characterRepository = characterRepository;
+        this.dataLoadOnStartup = dataLoadOnStartup;
+    }
+
+    @PostConstruct
+    @Override
+    public void saveCharactersToDB() {
+        List<CharacterResultDto> characterResultDtoList = dataLoadOnStartup.getCharacters();
+        for (CharacterResultDto ch : characterResultDtoList) {
+            characterRepository.save(convertDtoToCharacter(ch));
+        }
+    }
+
+    @Override
+    public Character convertDtoToCharacter(CharacterResultDto characterResultDto) {
+        Character character = new Character();
+        character.setExternalId(characterResultDto.getExternalId());
+        character.setName(characterResultDto.getName());
+        character.setStatus(characterResultDto.getStatus());
+        character.setGender(characterResultDto.getGender());
+        return character;
+    }
+
+    @Override
+    public Character getRandomCharacter() {
+        List<Character> characters = characterRepository.findAll();
+        Random random = new Random();
```

Comment:
We are using the spring, so instantiation of service classes via `new` is bad practice. 

Make a bean of random and autowire it. 
```suggestion
        Random random = new Random();
```